src/Observability/Runtime/Common/ExportFormatter.cs

```diff
@@ -148,6 +161,52 @@ public static string FormatLogData(IDictionary<string, object?> data)
             return resource.Attributes.FirstOrDefault(a => a.Key == "service.version").Value?.ToString();
         }
 
+        private static OtlpSpan BuildOtlpSpanWithTruncation(Activity activity, Action<string>? logInformation = null)
```

Comment:
nit: we can do this in a follow up PR but I think passing a delegate for logging here is an antipattern. What if we make this class instantiable and it gets the logger from DI like any other class. That way we can reduce the complexity of the arguments on all these methods and the need for null check on the `log*` calls.